DotNetDBF/DBFReader.cs

```diff
@@ -245,175 +244,156 @@ internal Object[] NextRecord(IEnumerable<int> selectIndexes, IList<int> sortedIn
+                                        int.TryParse(monthString, out month) &&
+                                        int.TryParse(dayString, out day))
+                                    {
+                                        recordObjects[i] = new DateTime(year, month, day);
+                                    }
+                                    else
+                                    {
+                                        recordObjects[i] = null;
+                                    }
                                 }
-                                else
+                                catch (ArgumentOutOfRangeException)
                                 {
+                                    /* this field may be empty or may have improper value set */
                                     recordObjects[i] = null;
                                 }
-                            }
-                            catch (ArgumentOutOfRangeException)
-                            {
-                                /* this field may be empty or may have improper value set */
-                                recordObjects[i] = null;
-                            }
-
-                            break;
 
+                                break;
+                            }
                         case NativeDbType.Float:
-
-                            try
                             {
-                                byte[] t_float = new byte[
-                                    _header.FieldArray[i].FieldLength
-                                    ];
-                                _dataInputStream.Read(t_float, 0, t_float.Length);
-                                String tParsed = CharEncoding.GetString(t_float);
-                                var tLast = tParsed.Substring(tParsed.Length - 1);
-                                if (tParsed.Length > 0
-                                    && tLast != " "
-                                    && tLast != DBFFieldType.Unknown)
+                                try
                                 {
-                                    recordObjects[i] = Double.Parse(tParsed, NumberStyles.Float | NumberStyles.AllowLeadingWhite);
+                                    byte[] t_float = new byte[_header.FieldArray[i].FieldLength];
+                                    _dataInputStream.Read(t_float, 0, t_float.Length);
+                                    string tParsed = CharEncoding.GetString(t_float);
+                                    string tLast = tParsed.Substring(tParsed.Length - 1);
+                                    if (tParsed.Length > 0
+                                        && tLast != " "
+                                        && tLast != DBFFieldType.Unknown)
+                                    {
+                                        recordObjects[i] = Convert.ToSingle(tParsed, NumberFormatProvider);

```

Comment:
So, you suggest to instead of property `NumberFormatProvider` create property `NumberFormatProviderInvariantInfo` of boolean type, which will be setting format provider to invariant?
Honestly, I don't quite understand what's the problem with "someone can change it" because I think the nature of format provider is the same as char encoding. Also, having whole HumberFormatInfo property is more flexible from the point of user as he can set any needed format.